coriolis/tests/osmorphing/test_ubuntu.py

```diff
@@ -0,0 +1,176 @@
+            mock.sentinel.operation_timeout)
+
+    @mock.patch.object(
+        base.BaseLinuxOSMorphingTools, '_version_supported_util'
+    )
+    def test_check_os_supported_not_supported(
+            self, mock_version_supported_util):
+        self.detected_os_info['distribution_name'] = 'unsupported'
+
+        result = ubuntu.BaseUbuntuMorphingTools.check_os_supported(
+            self.detected_os_info)
+
+        mock_version_supported_util.assert_not_called()
+
+        self.assertFalse(result)
+
+    @mock.patch.object(
+        base.BaseLinuxOSMorphingTools, '_version_supported_util'
+    )
+    def test_check_os_supported_lts_release(self, mock_version_supported_util):
+        result = ubuntu.BaseUbuntuMorphingTools.check_os_supported(
+            self.detected_os_info)
+
+        mock_version_supported_util.assert_called_once_with(
+            self.detected_os_info['release_version'], minimum=12.04,
+            maximum=12.04)
+
+        self.assertTrue(result)
+
+    @mock.patch.object(
+        base.BaseLinuxOSMorphingTools, '_version_supported_util'
+    )
+    def test_check_os_supported_non_lts_release(
+            self, mock_version_supported_util):
+        self.detected_os_info['release_version'] = '21.10'
+        mock_version_supported_util.return_value = False
+
+        result = ubuntu.BaseUbuntuMorphingTools.check_os_supported(
+            self.detected_os_info)
+
+        mock_version_supported_util.assert_called_with(
+            self.detected_os_info['release_version'], minimum=22.04)
+
+        self.assertEqual(result, mock_version_supported_util.return_value)
+
+    @mock.patch('yaml.dump')
+    @mock.patch.object(base.BaseLinuxOSMorphingTools, '_exec_cmd')
+    @mock.patch.object(base.BaseLinuxOSMorphingTools, '_write_file_sudo')
+    @mock.patch.object(base.BaseLinuxOSMorphingTools, '_read_file')
+    @mock.patch.object(base.BaseLinuxOSMorphingTools, '_list_dir')
+    def test__set_netplan_ethernet_configs(
+            self, mock_list_dir, mock_read_file, mock_write_file_sudo,
+            mock_exec_cmd, mock_yaml_dump):
+        mock_list_dir.return_value = ['file1', 'file2.yaml']
+        mock_read_file.return_value = (
+            b'network: {version: 2, ethernets: {eth0: {dhcp4: true}}}')
+        nics_info = [
+            {'name': 'eth0', 'mac_address': '00:00:00:00:00:00'},
+            {'name': 'eth1', 'mac_address': '00:00:00:00:00:01'}]
+
+        self.morphing_tools._set_netplan_ethernet_configs(
+            nics_info, dhcp=True, iface_name_prefix='test')
+
+        mock_exec_cmd.assert_called_once_with(
+            "sudo cp '%s/etc/netplan/file2.yaml' "
+            "'%s/etc/netplan/file2.yaml.bak'" % (
+                self.os_root_dir, self.os_root_dir))
+        mock_read_file.assert_called_once_with('etc/netplan/file2.yaml')
+        mock_write_file_sudo.assert_called_once_with(
+            'etc/netplan/file2.yaml', mock_yaml_dump.return_value)
+
+    @mock.patch('yaml.load')
```

Comment:
The test can be written without mocking this, but you have to adjust the inputs / assertions.